### DISCUSSION TITLE
CM15A Support?

### DISCUSSION THREAD

**Discussion by JeepRat2001** [HeyuX10Automation/heyu#43] [Q&A] [answered]
Hello, I've been trying to get "mocad" to work on rasberriPI4 but it is a real pain. I would rather use heyu, if it's possible to use a CM15A on usb. Is it possible to do this?

Steve

**Comment by JeepRat2001** [comment]
Or, can rs232 to usb adapters work the other way? Usually they're used to go from usb out to rs232 in. Could the RasberryPI serial output (which heyu can see) be used to interface the usb CM15A?

**Reply by arnieSkyNet** [reply]
I’m using on a raspberry pi3b with usb of that helps

**Comment by davepet** [comment, marked as answer]
My RPi3b uses a "Plugable" usb to serial adapter.
I connect both CM-15 and CM-17a and they work great with Heyu and Node-Red.
CM-17a alone gives an error because it doesn't respond like CM15 but it works.
I leave both connected so no problems.
Even works thru Alexa using a free account on Node-Red Smarthome.

**Reply by arnieSkyNet** [reply]
Cool. I use the habridge Java to control the heyu from Alexa as that keeps stuff going during outages 

https://github.com/bwssytems/ha-bridge

**Reply by davepet** [reply]
I used HA-bridge awhile back. It quit and I gave up on it.

**Reply by jkrzyszt** [reply]
> My RPi3b uses a "Plugable" usb to serial adapter. I connect both CM-15 and CM-17a and they work great with Heyu and Node-Red.

I think that answer is (unintentionally) misleading.  Support for CM15A has never been implemented in Heyu, and I would be really surprised if that worked. I think @davepet had a CM11A or a variant of it on mind, unless he was using a different piece of software for controlling his CM15A and that software was exchanging information with Heyu, e.g. over that Node-Red (which I know nothing about).

**Comment by JeepRat2001** [comment]
I've never tried Node-RED. What's a "plugable" usb to serial adapter? I have an old adapter cable that did that, but can you access the cm15a via usb directly with Node-RED?

**Comment by davepet** [comment]
Node-Red is great for my dabbling (Beginner-retired).
Plugable is the label on the device. It's one of these (https://plugable.com/products/pl2303-db9)
According to my Google search you can control USB with N-R.
I let Heyu do that.
